Tensile/KernelWriter.py

```diff
@@ -964,41 +1110,53 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
         localReads = Code.Module()
         localReadsA = Code.Module()
         localReadsB = Code.Module()
-        for iui in range(0,kernel["InnerUnroll"]):
-          if self.enable["LocalRead"]:
-            if kernel["MatrixInstruction"] and \
-              (kernel["ProblemType"]["DataType"].isBFloat16() or kernel["ProblemType"]["DataType"].isHalf()) and \
-              not kernel["TransposeLDS"] :
-              # Reading 16-bit data from LDS requires packing when ECC enabled
-              localReads.addText(self.comment("local read a"))
-              localReadCodeA, packCodeA = self.localReadDo(kernel, plrIdx, iui, 0, ((u+pflr)*kernel["InnerUnroll"]+iui), tensorParametersA)
-              localReads.addCode(localReadCodeA)
-              localReads.addText(self.comment("local read b"))
-              localReadCodeB, packCodeB = self.localReadDo(kernel, plrIdx, iui, 0, ((u+pflr)*kernel["InnerUnroll"]+iui), tensorParametersB)
-              localReads.addCode(localReadCodeB)
-              localReadsA.addCode(localReadCodeA)
-              localReadsB.addCode(localReadCodeB)
-              pack[plrIdx].addCode(packCodeA)
-              pack[plrIdx].addCode(packCodeB)
-            else:
-              localReads.addText(self.comment("local read a"))
-              localReads.addCode(self.localReadDo(kernel, plrIdx, iui, 0,((u+pflr)*kernel["InnerUnroll"]+iui), tensorParametersA))
-              localReads.addText(self.comment("local read b"))
-              localReads.addCode(self.localReadDo(kernel, plrIdx, iui, 0,((u+pflr)*kernel["InnerUnroll"]+iui), tensorParametersB))
-              #container for holding local read A & B elements for later re-ordering
-              localReadsA.addCode(self.localReadDo(kernel, plrIdx, iui, 0,((u+pflr)*kernel["InnerUnroll"]+iui), tensorParametersA))
-              localReadsB.addCode(self.localReadDo(kernel, plrIdx, iui, 0,((u+pflr)*kernel["InnerUnroll"]+iui), tensorParametersB))
-
-            # Don't increment the LRO if we are going to reset them below:
-            if not isResetLroIter or iui != kernel["InnerUnroll"]-1:
-              localReads.addText(self.comment("local read increment a"))
-              localReads.addText(self.localReadInc(kernel, iui, tensorParametersA))
-              localReads.addText(self.comment("local read increment b"))
-              localReads.addText(self.localReadInc(kernel, iui, tensorParametersB))
+        # if not PrefetchGlobalRead, PrefetchLocalRead is at start of the loop
+        if (u < kernel["LoopIters"] - kernel["PrefetchLocalRead"]) or kernel["PrefetchGlobalRead"]:
```

Comment:
add this if statement,
original code only consider prefetchLocalRead=0 or 1 in "unrolled loop: last summation iter" part.
However, if prefetchLocalRead>1, we also need to consider to add localread or not in "unroll loop:" part.